src/main.ts

```diff
@@ -1,9 +1,37 @@
 import { Plugin, MarkdownView, Editor } from "obsidian";
+import {CheckboxReorderPluginSettingTab} from './settings'
 import { reorderCheckboxesInFile } from "src/reorderCheckboxes";
 
+interface CheckboxReorderPluginSettings {
+	autoReorder: boolean;
+}
+
+const DEFAULT_SETTINGS: Partial<CheckboxReorderPluginSettings> = {
+	autoReorder: false
+}
+
 export default class CheckboxReorderPlugin extends Plugin {
+	settings: CheckboxReorderPluginSettings;
+
 	async onload() {
+		await this.loadSettings();
+
 		console.log("Checkbox Reorder Plugin loaded");
+		
+
+		this.addSettingTab(new CheckboxReorderPluginSettingTab(this.app, this))
+
+		this.registerEvent(this.app.vault.on("modify", () => {
```

Comment:
this look a little strange but could be right nontheless `this.registerEvent(this.app.vault.on(...`

Reply:
It should be correct...
https://docs.obsidian.md/Plugins/Events